MREUnityRuntime/MREUnityRuntimeLib/Assets/AssetLoader.cs

```diff
@@ -78,7 +78,29 @@ internal IList<Actor> CreateFromPrefab(Guid prefabId, Guid? parentId)
 			var actorList = new List<Actor>();
 			MWGOTreeWalker.VisitTree(instance, go =>
 			{
-				go.layer = UnityConstants.ActorLayerIndex;
+				if (go.GetComponent<UnityEngine.Collider>() != null && collisionLayer != null)
```

Comment:
>(go.GetComponent<UnityEngine.Collider>() != null [](start = 7, length = 48)

Not sure I completely understand why it has to have a collider to be assigned a layer, or even if I agree with this.  Why are we explicitly not allowing them to assign a layer to a game object when creating from a prefab at this point in the create process.  It seems unnecessarily restrictive as they could early assign a layer to any of the created prefabs and simply add a different collider type to each after the creation without having to know or assign the layer later on in the code.

Is this an attempt to remove the Unity like behavior that layers are associated with game objects rather than colliders?

Reply:
Your last statement, pretty much. Unity uses layers for lots of different things, but we only use it for collisions. As such, it seemed to make sense to put the layer on the collider, like other engines do. There's no harm in putting collider-less actors on a layer, but it doesn't serve any purpose either. For consistency, I try to keep collider-less actors in the Default layer.

---
In reply to: [335525126](https://github.com/microsoft/mixed-reality-extension-unity/pull/119#discussion_r335525126) [](ancestors = 335525126)

Reply:
I guess that is fine.

---
In reply to: [335597299](https://github.com/microsoft/mixed-reality-extension-unity/pull/119#discussion_r335597299) [](ancestors = 335597299,335525126)